SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,501 @@
+
+			ViewCoordinator& Coordinator = NetDriver->Connection->GetCoordinator();
+			Coordinator.SendCreateEntityRequest(MoveTemp(SkeletonEntityManifestComponents), /*EntityId =*/{});
+
+			UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Created manifest for VirtualWorker %d WorkerPartition %lld"), WorkerId,
+				   ServerWorkerPartitionId);
+		}
+
+		ManifestsCreatedCount = WorkerToEntitiesToPopulate.Num();
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Finished creating manifests"));
+		Stage = EStage::WaitingForPopulation;
+	}
+
+	if (Stage == EStage::WaitingForPopulation)
+	{
+		auto ProcessEntities = [](bool& bIsFirstCall, const FSubView& SubView, const TFunction<void(Worker_EntityId)> Handler) {
+			if (bIsFirstCall)
+			{
+				bIsFirstCall = false;
+				for (const Worker_EntityId EntityId : SubView.GetCompleteEntities())
+				{
+					Handler(EntityId);
+				}
+			}
+			else
+			{
+				for (const EntityDelta& Delta : SubView.GetViewDelta().EntityDeltas)
+				{
+					checkf(Delta.Type != EntityDelta::REMOVE,
+						   TEXT("Skeleton entity manifest must never be removed, but was removed from entity %lld"), Delta.EntityId)
+
+						if (Delta.Type == EntityDelta::ADD || Delta.Type == EntityDelta::UPDATE
+							|| Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
+					{
+						const Worker_EntityId EntityId = Delta.EntityId;
+						Handler(EntityId);
+					}
+				}
+			}
+		};
+
+		auto ProcessManifest = [this](const Worker_EntityId ManifestEntityId) {
+			const EntityViewElement* Element = ManifestSubView->GetView().Find(ManifestEntityId);
+
+			const ComponentData* ManifestComponent =
+				Element->Components.FindByPredicate(ComponentIdEquality{ FSkeletonEntityManifest::ComponentId });
+
+			checkf(ManifestComponent != nullptr, TEXT("Skeleton entity manifest must never be removed, but was removed from entity %lld"),
+				   ManifestEntityId);
+
+			const FSkeletonEntityManifest Manifest(*ManifestComponent);
+
+			Manifests.Add(ManifestEntityId, Manifest);
+		};
+
+		ProcessEntities(bIsFirstTimeProcessingManifests, *ManifestSubView, ProcessManifest);
+
+		if (Manifests.Num() == ManifestsCreatedCount)
+		{
+			const bool bAllManifestsFinished =
+				Algo::AllOf(Manifests, [](const TPair<Worker_EntityId_Key, FSkeletonEntityManifest>& ManifestPair) {
+					return ManifestPair.Value.PopulatedEntities.Num() == ManifestPair.Value.EntitiesToPopulate.Num();
+				});
+
+			if (bAllManifestsFinished)
+			{
+				Stage = EStage::FinalizingSkeletons;
+			}
+		}
+	}
+
+	if (Stage == EStage::FinalizingSkeletons)
+	{
+		for (const Worker_EntityId_Key SkeletonEntityId : CreatedSkeletonEntities)
+		{
+			ViewCoordinator& Coordinator = NetDriver->Connection->GetCoordinator();
+
+			Coordinator.SendRemoveComponent(SkeletonEntityId, SpatialConstants::FLESHOUT_REQUIRED_TAG_COMPONENT_ID, {});
+			Coordinator.SendRemoveComponent(SkeletonEntityId, SpatialConstants::FLESHOUT_QUERY_TAG_COMPONENT_ID, {});
+		}
```

Comment:
It's either removing these tags, or adding a new tag that overrides these (FLESHOUT_FINISHED_TAG), or not adding ACTOR_TAG and the rest until population is finished; the last approach is the cleanest one as it avoids changes to Actor/LB subviews' filters, but otherwise seems to require refactors.

Reply:
We had a discovery about the guarantees we will have regarding adding adding dynamic components, namely that added components will be observed in the same order than they have been added. Do we have the same guarantees for removed components?

Reply:
Yes we've discussed this with @samiwh; component removals aren't ordered relative to component additions, which means there's a race here. Adding a FLESHOUT_FINISHED tag instead would solve this, but this *should* be fine for the spike.

Reply:
Probably why you were having issues with the RPC components though.

Reply:
The guarantees are (going to be) that component additions happen broadly in the order you expect and that removals are basically unordered. Although worth noting that this is currently how we think things behave it's not "guaranteed" yet.

But it means that yes we need a finished tag. So you flesh things out if you see a fleshout tag and not a finished tag. You could then remove the fleshout tag if that is actually an optimistaion.